indico/modules/events/editing/controllers/backend/management.py

```diff
@@ -176,3 +179,33 @@ def _process_POST(self, principals):
         for p in old_principals - principals:
             self.event.update_principal(p, del_permissions={permission_name})
         return '', 204
+
+
+class RHDownloadArchive(RHEditablesBase):
+    def _process(self):
+        return generate_editables_zip(self.editables)
+
+
+class RHAssignEditor(RHEditablesBase):
+    def _process(self):
+        argmap = {
+            'editor': Principal(required=True)
```

Comment:
You can use the `use_kwargs` decorator for this

Reply:
Ah.. true.. I should have done that. Initially I was passing some redundant objects to `EditableList` and that is why I didn't use available decorators. But after some time I realized I don't need to pass anything like `self.event` or `self.contribution`.

Reply:
Actually, you do! You need to ensure that you cannot pass editable ids which do not belong to the given event!

So, the decorator may not work after all unless you can do it with the context coming from the RH..